test/distributed/test_distributed.py

```diff
@@ -3130,6 +3131,47 @@ def test_broadcast_object_list(self):
         dist.broadcast_object_list(objects, src=0)
         self.assertEqual(objects, collectives_object_test_list)
 
+    @require_backend({"nccl", "gloo"})
+    @require_n_gpus_for_nccl_backend(int(os.environ["WORLD_SIZE"]), os.environ["BACKEND"])
```

Comment:
Can we also add some unit test that directly test the `scatter` API with a variety of test cases? For example the case for `namedtuple(a=1, b=1)` you mentioned in the docs is a good test.

Reply:
I added this test in `test/test_cuda.py` since it doesn't look like there were tests for this API anywhere else.